src/client/Client.js

```diff
@@ -468,8 +470,8 @@ class Client extends BaseClient {
       throw new TypeError('CLIENT_INVALID_OPTION', 'shards', "'auto', a number or array of numbers");
     }
     if (options.shards && !options.shards.length) throw new RangeError('CLIENT_INVALID_PROVIDED_SHARDS');
-    if (typeof options.messageCacheMaxSize !== 'number' || isNaN(options.messageCacheMaxSize)) {
-      throw new TypeError('CLIENT_INVALID_OPTION', 'messageCacheMaxSize', 'a number');
+    if (typeof options.makeCache !== 'function') {
```

Comment:
Otherwise this throws when it's unset:
```suggestion
    if (typeof options.makeCache !== 'undefined && typeof options.makeCache !== 'function') {
```

Reply:
It should throw when unset, it needs to be set (either via defaults or user) otherwise caching wouldn't work at all (probably hard crash)

Reply:
Wait, nevermind, the options are merged beforehand.